libretro/libretro.cpp

```diff
@@ -1233,8 +1233,11 @@ namespace Libretro
       emuThreadState = EmuThreadState::QUIT_REQUESTED;
 
       // Need to keep eating frames to allow the EmuThread to exit correctly.
-      while (ctx->ThreadFrame())
-         ;
+      // The default implementation of GraphicsContext::ThreadFrame always returns true.
+      // The only LibretroGraphicsContexts that override this behavior are the GL ones.
+      if (ctx->GetGPUCore() == GPUCORE_GLES)
```

Comment:
It's better that we not hardcode this assumption.  These may change and people might be confused about why those methods are not working - and only not working on i.e. Vulkan or Direct3D.

If you really think this is important, it'd be best to add a comment in those methods saying they're nitentionally not called (and why.)  But there should be motivation, and I'm not sure I see it here?

-[Unknown]